src/rust/scheduler/handle.rs

```diff
@@ -32,7 +32,9 @@ use ::std::{
 #[derive(Clone)]
 pub struct SchedulerHandle {
     /// Corresponding location in the scheduler's memory chunk.
-    key: Option<u64>,
+    offset: Option<u64>,
+    /// External identifying token.
+    token: Option<u64>,
```

Comment:
I think this would introduce a circular dependency here between the scheduler and runtime. I would think of QToken as being a wrapper around the scheduler token. I could perhaps call it task_id for more clarity.

Reply:
Well, I was talking about making `scheduler` to own and expose `QToken`. Alternatively, we could introduce a scheduler type like `SchedulerToken` and make `QToken` to wrap it. The point here is exactly making the compiler complain if one assigns a `u64` that concerns a token to a `u64` that concerns offset (and vice versa).